### PR TITLE
support tiflash_fastscan variable from tidb, remove fast mode in table_info, and change name fast mode to fast scan

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #5252 

Problem Summary:

### What is changed and how it works?

1. remove the code related to fast_mode in table_info
2. support is_fast_scan from TableScan and PartitionTableScan from TiDB
3. change the name 'fast mode' to 'fast scan'

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/fullstack-test2/variables/set_variable_fastscan.test [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+mysql> create table test.t (a int, b int)
+mysql> alter table test.t set tiflash replica 1
+
+func> wait_table test t
+
+mysql> insert into test.t values(1,2);
+mysql> insert into test.t values(2,4);
+mysql> insert into test.t values(5,7);
+
+mysql> set @@tiflash_fastscan=ON; set session tidb_isolation_read_engines='tiflash'; select * from test.t;
```
Comment (breezewish) [inline tests/fullstack-test2/variables/set_variable_fastscan.test:26]: Both are session variables, better to use the same style:

```suggestion
mysql> set @@tiflash_fastscan=ON; set @@tidb_isolation_read_engines='tiflash'; select * from test.t;
```

File: tests/fullstack-test2/variables/set_variable_fastscan.test [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
++------+-----+
+| a    | b   |
++------+-----+
+|    2 |    4|
+|    5 |    7|
++------+-----+
+
+mysql> alter table test.t compact tiflash replica;
+
+mysql> set @@tiflash_fastscan=ON; set session tidb_isolation_read_engines='tiflash'; select * from test.t;
```
Comment (breezewish) [inline tests/fullstack-test2/variables/set_variable_fastscan.test:56]: Is it possible to test multiple sessions? For example, if you incorrectly implemented this session variable to be effective globally, this test case will still pass.

**Review by breezewish — APPROVED** [review]
Some minor test case suggestions.

File: tests/fullstack-test2/variables/set_variable_fastscan.test (hunk 2 above)
Comment (hongyunyan) [reply tests/fullstack-test2/variables/set_variable_fastscan.test]: Yes, I add more comprehensive test cases here.

File: dbms/src/Storages/SelectQueryInfo.h [hunk 3]
```diff
@@ -14,6 +14,8 @@
 
 #pragma once
 
+#include <Flash/Coprocessor/TiDBTableScan.h>
```
Comment (lidezhu) [inline dbms/src/Storages/SelectQueryInfo.h:17]: Seems this header is not needed?

File: dbms/src/Storages/StorageDeltaMerge.cpp [hunk 4]
```diff
@@ -758,7 +759,7 @@ BlockInputStreams StorageDeltaMerge::read(
         rs_operator,
         query_info.req_id,
         query_info.keep_order,
-        /* is_fast_mode */ tidb_table_info.tiflash_mode == TiDB::TiFlashMode::Fast, // read in normal mode or read in fast mode
+        /* is_fast_scan */ query_info.is_fast_scan, // read in normal mode or read in fast mode
```
Comment (lidezhu) [inline dbms/src/Storages/StorageDeltaMerge.cpp:762]:
```suggestion
        /* is_fast_scan */ query_info.is_fast_scan,
```

File: dbms/src/Storages/StorageDeltaMerge.cpp [hunk 5]
```diff
@@ -24,6 +24,7 @@
 #include <DataTypes/isSupportedDataTypeCast.h>
 #include <Databases/IDatabase.h>
 #include <Debug/MockTiDB.h>
+#include <Flash/Coprocessor/TiDBTableScan.h>
```
Comment (lidezhu) [inline dbms/src/Storages/StorageDeltaMerge.cpp:27]: Remove this header if not needed.

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.h [hunk 6]
```diff
@@ -359,8 +359,8 @@ class DeltaMergeStore : private boost::noncopyable
 
 
     /// Read rows in two modes:
-    ///     when is_fast_mode == false, we are in normal mode. Thus we will read rows with MVCC filtering, del mark !=0  filter and sorted merge
-    ///     when is_fast_mode == true, we are in fast mode. Thus we will read rows without MVCC and sorted merge
+    ///     when is_fast_scan == false, we are in normal mode. Thus we will read rows with MVCC filtering, del mark !=0  filter and sorted merge
```
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.h:362]:
```suggestion
    ///     when is_fast_scan == false, we will read rows with MVCC filtering, del mark !=0  filter and sorted merge.
```

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.h [hunk 7]
```diff
@@ -359,8 +359,8 @@ class DeltaMergeStore : private boost::noncopyable
...
+    ///     when is_fast_scan == true, we are in fast mode. Thus we will read rows without MVCC and sorted merge
```
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.h:363]:
```suggestion
    ///     when is_fast_scan == true, we will read rows without MVCC and sorted merge.
```

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.h [hunk 8]
```diff
@@ -371,7 +371,7 @@ class DeltaMergeStore : private boost::noncopyable
                            const RSOperatorPtr & filter,
                            const String & tracing_id,
                            bool keep_order,
-                           bool is_fast_mode = false, // set true when read in fast mode
+                           bool is_fast_scan = false, // set true when read in fast mode
```
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.h:374]:
```suggestion
                           bool is_fast_scan = false,
```

File: dbms/src/Storages/DeltaMerge/File/DMFileBlockInputStream.h [hunk 9]
```diff
@@ -92,11 +92,11 @@ class DMFileBlockInputStreamBuilder
     // Only set is_fast_mode_ param to true when read in fast mode.
     // Only set enable_del_clean_read_ param to true when you don't need del columns in fast mode.
     // `max_data_version_` is the MVCC filter version for reading. Used by clean read check
-    DMFileBlockInputStreamBuilder & enableCleanRead(bool enable_handle_clean_read_, bool is_fast_mode_, bool enable_del_clean_read_, UInt64 max_data_version_)
+    DMFileBlockInputStreamBuilder & enableCleanRead(bool enable_handle_clean_read_, bool is_fast_scan_, bool enable_del_clean_read_, UInt64 max_data_version_)
```
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/File/DMFileBlockInputStream.h:95]: Please change the code comment.

**Review by lidezhu — APPROVED** [review]
LGTM

**Pull request merged**